common/pubsub/pubsub_async_test.go

```diff
@@ -0,0 +1,255 @@
+package pubsub_test
+
+import (
+	"context"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/status-im/status-go/common/pubsub"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPubSubAsync_Int(t *testing.T) {
+	ctx := context.Background()
+	publisher := pubsub.NewPublisher()
+	defer publisher.Close()
+
+	testingCh, unsub := pubsub.SubscribeAsync[int](publisher)
+	defer unsub()
+
+	val := 42
```

Comment:
```suggestion
	val := gofakeit.Int()
```

(it seems you'd need to upgrade `gofakeit` to v7 to get nice `Int()` though)

Reply:
ahh cool, I'll use that instead